### PR TITLE
Add developer docs for sdkx-cocos2dx Android

### PR DESCRIPTION
**Pull request by helpshift-tanuj**

### CODE REVIEW

File: docs/sdkx-cocos2dx/deep-linking.mdx [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+---
+
+import {
+  Admonition,
+  TabItem,
+  Image,
+  Intro,
+} from "@site/src/components/forDocs";
+
+# Deep Linking in FAQs {#android-deep-links}
```
Comment (sumeet-helpshift) [inline docs/sdkx-cocos2dx/deep-linking.mdx:14]: The link shouldnt contain "android" i guess

File: docs/sdkx-cocos2dx/delegates.mdx [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+bool HelloWorld::init()
+{
+    //////////////////////////////
+    // 1. super init first
+    if ( !Scene::init() )
+    {
+        return false;
+    }
+
+    HelpshiftCocos2dx::setHelpshiftEventsListener(handleHelpshiftEvent,
```
Comment (sumeet-helpshift) [inline docs/sdkx-cocos2dx/delegates.mdx:62]: there should be code for install call before this so that it clear

File: docs/sdkx-cocos2dx/getting-started.mdx [hunk 3]
```diff
@@ -0,0 +1,308 @@
...
+
+If your game does not use this permission, you would need to add this key as well as description for the same. Not adding this key-description pair might cause game rejection.
+
+**NSPhotoLibraryUsageDescription Description text:** "We need to access photos library to allow users manually pick images meant to be sent as attachment for help and support reasons."
+
+</Admonition>
+
+<Admonition type="info" title="Note">
+
+The following instructions apply only to **versions below SDK X 10.2.0**.
```
Comment (sumeet-helpshift) [inline docs/sdkx-cocos2dx/getting-started.mdx:123]: do we need this section in Cocos plugin ?
It will be at or above 10.2.0 right ?

File: docs/sdkx-cocos2dx/getting-started.mdx [hunk 4]
```diff
@@ -0,0 +1,308 @@
...
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // Add install configs in the config map
+        Map<String, Object> config = new HashMap<>();
+        //...
+        // Install call
+        HelpshiftCocosBridge.install(
```
Comment (sumeet-helpshift) [inline docs/sdkx-cocos2dx/getting-started.mdx:199]: Are we not suggesting install call from C++ layer for android ?

File: docs/sdkx-cocos2dx/migration-guide.mdx [hunk 5]
```diff
@@ -0,0 +1,635 @@
...
+- [FAQ filtering by tags](/cocos2d-x/support-tools-android/#faq-filtering)
+- [Minimum Issue description length](/cocos2d-x/sdk-configuration-android/#minimum-issue-desc-length)
+- [SDK UI strings customization](/cocos2d-x/string-customization-android/)
+- [Switching between light and dark themes](/cocos2d-x/design-android/#skinning)
+
+These features are yet to be developed, and we may add them to our future roadmap. You can reach out to [support@helpshift.com](mailto:support@helpshift.com) to raise any query for the above.
+
+## Supported SDK Versions {#supported-sdk-versions}
+
+A successful migration indicates maintaining users’ data and chat history when they move from one SDK version to another. We support the migration of the SDK Cocos2dx plugin from version 4.x+ to the SDK X Cocos2dx plugin.
```
Comment (sumeet-helpshift) [inline docs/sdkx-cocos2dx/migration-guide.mdx:61]: We should try to clearly mention that 4.x is Legacy SDK Cocos2dx plugin version and not Cocos2dx version itself.

File: docs/sdkx-cocos2dx/migration-guide.mdx [hunk 6]
```diff
@@ -0,0 +1,635 @@
...
+
+<Admonition type="info" title="Note">
+
+** Migration from SDK Cocos2dx plugin version < 3.x **
+
+We don’t support the migration from SDK Cocos2dx plugin version < 4.x. If you move to SDK X Cocos2dx plugin from version < 4.x, SDK X will function correctly, but the logged-in user information will be lost, and a new default user will be created. You can log the current user in again, but the corresponding data for that user will not be migrated. It will be treated as a new user in the Helpshift system.
```
Comment (sumeet-helpshift) [inline docs/sdkx-cocos2dx/migration-guide.mdx:67]: Same here

File: docs/sdkx-cocos2dx/migration-guide.mdx [hunk 7]
```diff
@@ -0,0 +1,635 @@
...
+
+with SDK X code -
+
+```cpp
+@Override
+public void onMessageReceived(RemoteMessage message) {
+  Map<String, String> data = message.getData();
+  String origin = data.get("origin");
+  if (origin != null && origin.equals("helpshift")) {
+    HelpshiftCocosBridge.handlePushNotification(getApplicationContext(), data);
```
Comment (sumeet-helpshift) [inline docs/sdkx-cocos2dx/migration-guide.mdx:342]: there should be install call first here i think

File: docs/sdkx-cocos2dx/notifications-android.mdx [hunk 8]
```diff
@@ -0,0 +1,264 @@
...
+
+2. Using FCM's new `FirebaseMessagingService` which provides "RemoteMessage" argument, you should use `HelpshiftCocosBridge.handlePush` API which takes a Map argument.
+
+```java
+@Override
+public void onMessageReceived(RemoteMessage message) {
+  Map<String, String> data = message.getData();
+  String origin = data.get("origin");
+  if (origin != null && origin.equals("helpshift")) {
+      HelpshiftCocosBridge.handlePushNotification(getApplicationContext(), data);
```
Comment (sumeet-helpshift) [inline docs/sdkx-cocos2dx/notifications-android.mdx:93]: should we include install call here ?

File: docs/sdkx-cocos2dx/notifications-android.mdx [hunk 9]
```diff
@@ -0,0 +1,264 @@
...
+
+public class MyActivity extends Cocos2dxActivity {
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // Add largeNotificationIcon config in install config map
+        Map<String, Object> config = new HashMap<>();
+        config.put("notificationIcon", "icon-image-file-name");
```
Comment (sumeet-helpshift) [inline docs/sdkx-cocos2dx/notifications-android.mdx:131]: pls check if we take file names or resources Ids here ?

File: docs/sdkx-cocos2dx/notifications-android.mdx [hunk 10]
```diff
@@ -0,0 +1,264 @@
...
+
+public class MyActivity extends Cocos2dxActivity {
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // Add notificationSound config in install config map
+        Map<String, Object> config = new HashMap<>();
+        config.put("notificationSound", "<notification-sound-file-name>");
```
Comment (sumeet-helpshift) [inline docs/sdkx-cocos2dx/notifications-android.mdx:161]: same

File: docs/sdkx-cocos2dx/notifications-android.mdx [hunk 11]
```diff
@@ -0,0 +1,264 @@
...
+            "<App Id from the Helpshift Dashboard>",
+            "<Domain name from the Helpshift Dashboard>",
+            config);
+    }
+}
+```
+
+## Showing notification count when replies are sent to the user {#showing-notification-count}
+
+To fetch unread messages count from the server you can call `requestUnreadMessageCount(boolean shouldFetchFromServer)` API. This API will return unread messages count via delegate.
```
Comment (sumeet-helpshift) [inline docs/sdkx-cocos2dx/notifications-android.mdx:237]: we should try to make it more clear that they need to register delegate listener for this.
Also make it clear in code that the impl shown is of delegate listener

File: docs/sdkx-cocos2dx/migration-guide.mdx (hunk 7 above)
Comment (helpshift-tanuj) [reply docs/sdkx-cocos2dx/migration-guide.mdx:342]: We do a `install` call with cached values inside `handlePushNotification` method.

File: docs/sdkx-cocos2dx/notifications-android.mdx (hunk 8 above)
Comment (helpshift-tanuj) [reply docs/sdkx-cocos2dx/notifications-android.mdx:93]: We do a `install` call with cached values inside `handlePushNotification` method.

File: docs/sdkx-cocos2dx/delegates.mdx (hunk 2 above)
Comment (helpshift-tanuj) [reply docs/sdkx-cocos2dx/delegates.mdx:62]: @sumeet-helpshift We dont allow install call in android from cocos layer.

File: docs/sdkx-cocos2dx/getting-started.mdx (hunk 3 above)
Comment (helpshift-tanuj) [reply docs/sdkx-cocos2dx/getting-started.mdx:123]: Yes, will remove this

File: docs/sdkx-cocos2dx/deep-linking.mdx (hunk 1 above)
Comment (helpshift-tanuj) [reply docs/sdkx-cocos2dx/deep-linking.mdx:14]: Will fix this

File: docs/sdkx-cocos2dx/getting-started.mdx (hunk 4 above)
Comment (helpshift-tanuj) [reply docs/sdkx-cocos2dx/getting-started.mdx:199]: No, In android `install` call can only be done from java layer as we expect an activity context

File: docs/sdkx-cocos2dx/sdk-configuration.mdx [hunk 12]
```diff
@@ -0,0 +1,486 @@
...
+  TabItem,
+  Image,
+  Intro,
+} from "@site/src/components/forDocs";
+
+# SDK Configuration
+
+<Intro>
+
+Helpshift provides several config options which can be used to customize behaviour of the SDK. Use these options in conjunction with our [Helpshift APIs](/sdkx_android/support-tools/).
```
Comment (sumeet-helpshift) [inline docs/sdkx-cocos2dx/sdk-configuration.mdx:19]: incorrect link to Helpshift APIs

File: docs/sdkx-cocos2dx/sdk-configuration.mdx [hunk 13]
```diff
@@ -0,0 +1,486 @@
...
+
+public class MyActivity extends Cocos2dxActivity {
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // Add screenOrientation config in install config map
+        Map<String, Object> config = new HashMap<>();
+        config.put("screenOrientation", 0);
```
Comment (sumeet-helpshift) [inline docs/sdkx-cocos2dx/sdk-configuration.mdx:286]: Since this is java code, we can use ActivityInfo class for orientation constants

File: docs/sdkx-cocos2dx/sdk-configuration.mdx [hunk 14]
```diff
@@ -0,0 +1,486 @@
...
+
+public class MyActivity extends Cocos2dxActivity {
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // Add notificationIcon config in install config map
+        Map<String, Object> config = new HashMap<>();
+        config.put("notificationIcon", "<icon-image-file-name>");
```
Comment (sumeet-helpshift) [inline docs/sdkx-cocos2dx/sdk-configuration.mdx:315]: check if we need name or the resource id of the icon file

File: docs/sdkx-cocos2dx/sdk-configuration.mdx [hunk 15]
```diff
@@ -0,0 +1,486 @@
...
+
+public class MyActivity extends Cocos2dxActivity {
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        // Add largeNotificationIcon config in install config map
+        Map<String, Object> config = new HashMap<>();
+        config.put("largeNotificationIcon", "<large-icon-image-file-name>");
```
Comment (sumeet-helpshift) [inline docs/sdkx-cocos2dx/sdk-configuration.mdx:345]: same

File: docs/sdkx-cocos2dx/tracking.mdx [hunk 16]
```diff
@@ -0,0 +1,215 @@
...
+
+You can attach Custom Issue Fields to every new conversation started by the user. A Custom Issue Field should have a key, a datatype, and a value. The Helpshift SDK allows the addition of Custom Issue Fields by using the `cifs` key in API config map.
+
+<Admonition type="info" title="Note">
+  We would like to inform you that the usage of the <b>"customIssueFields"</b>{" "}
+  key will be deprecated. We strongly recommend transitioning to using the{" "}
+  <b>"cifs"</b> key as the preferred method for passing custom issue fields.
+</Admonition>
+
+These custom issue fields will be sent whenever an end user starts a new conversation or creates an issue via Smart FAQs.
```
Comment (sumeet-helpshift) [inline docs/sdkx-cocos2dx/tracking.mdx:98]: why is this line needed for Smart FAQs specially ?

File: docs/sdkx-cocos2dx/tracking.mdx [hunk 17]
```diff
@@ -0,0 +1,215 @@
...
+
+For example:
+
+```cpp
+HelpshiftCocos2dx::clearBreadCrumbs();
+```
+
+<Admonition type="info" title="Note">
+
+- Applicable to SDK X v10.1.0 & above
```
Comment (sumeet-helpshift) [inline docs/sdkx-cocos2dx/tracking.mdx:189]: since we will package 10.2.0, is this note needed ?

File: docs/sdkx-cocos2dx/tracking.mdx [hunk 18]
```diff
@@ -0,0 +1,215 @@
...
+
+```cpp
+HelpshiftLog::d("<debug-tag>", "<debug-message>");
+```
+
+Debug logs are collected within the set debug logs limit. The limit is set under the **SDK Configurations** section for App Settings in the Helpshift’s Agent Dashboard. Debug logs are collected in a FIFO queue.
+
+<Admonition type="info" title="Note">
+
+- Applicable to SDK X v10.1.0 & above
```
Comment (sumeet-helpshift) [inline docs/sdkx-cocos2dx/tracking.mdx:210]: same here

File: docs/sdkx-cocos2dx/users.mdx [hunk 19]
```diff
@@ -0,0 +1,294 @@
...
+    <th>Parameter</th>
+    <th>Require/Optional</th>
+    <th>SDK Description</th>
+    <th>Inportant Considerations</th>
+  </tr>
+  <tr>
+    <td>userName</td>
+    <td>Optional</td>
+    <td>Provide the name you'd like your Agents to use when interacting with end users. If you don't know the user's name, you can leave it blank, and the Identity Bot (when enabled) will ask that user for their name. If you provide a value, the Identity Bot will not ask the user for their name again.</td>
+    <td>Max length: 255 characters. Values larger than this will be truncated.</td>
```
Comment (sumeet-helpshift) [inline docs/sdkx-cocos2dx/users.mdx:67]: we have released an update in this table recently.
Please refer to other plugins doc and update the entire content here accordingly.

File: docs/sdkx-cocos2dx/users.mdx [hunk 20]
```diff
@@ -0,0 +1,294 @@
...
+    ...
+}
+void handleAuthFailureEvent(HSAuthenticationFailureReason reason) {
+    if (reason == HSAuthenticationFailureReason::HSAuthTokenNotProvided) {
+        ...
+    } else if (reason == HSAuthenticationFailureReason::HSInvalidAuthToken) {
+        ...
+    }
+}
+HelpshiftCocos2dx::setHelpshiftEventsListener(handleHelpshiftEvent,
```
Comment (sumeet-helpshift) [inline docs/sdkx-cocos2dx/users.mdx:288]: i think this line is missing in all other places where we have used handleHelpshiftEvent or handleAuthFailureEvent.
Please check if we should be using this line at all other places for clarity in code samples

File: src/data/homeSections.json [hunk 21]
```diff
@@ -1,215 +1,223 @@
 [
-    {
-        "title": "Latest SDKs",
```
Comment (sumeet-helpshift) [inline src/data/homeSections.json:3]: there is huge change in this file due to tab/indents change.
Please ensure that everything is correct and nothing unintentional has changed.

File: src/data/navbar.json [hunk 22]
```diff
@@ -1,107 +1,111 @@
 {
-    "navbar": {
```
Comment (sumeet-helpshift) [inline src/data/navbar.json:2]: same here

File: src/data/sidebars.json [hunk 23]
```diff
@@ -1,520 +1,533 @@
 {
-    "sdkxAndroid": [
```
Comment (sumeet-helpshift) [inline src/data/sidebars.json:2]: same here

File: docs/sdkx-cocos2dx/notifications-android.mdx (hunk 9 above)
Comment (helpshift-tanuj) [reply docs/sdkx-cocos2dx/notifications-android.mdx:131]: Ye we expect resources ids, Will fix

File: docs/sdkx-cocos2dx/notifications-ios.mdx [hunk 24]
```diff
@@ -0,0 +1,263 @@
...
+Currently we support one flag i.e `enableInAppNotification`.
+
+### Enabling/Disabling In-app notifications {#enable-disable-in-app}
+
+<div className="compact ">
+
+|         |                         |
+| ------- | ----------------------- |
+| Flag    | enableInAppNotification |
+| Values  | true/false              |
```
Comment (jaydeep-helpshift) [inline docs/sdkx-cocos2dx/notifications-ios.mdx:190]: Use YES/NO rather than true/false here - or change the YES/NO in description that follows to true/false.

File: docs/sdkx-cocos2dx/outbound-support.mdx [hunk 25]
```diff
@@ -0,0 +1,253 @@
...
+
+You may want to add configuration specific to the current user in your app when they click on the notification.
+
+Setting local API config enables the Helpshift SDK to merge configuration from both, the config embedded in the outbound support link (as mentioned in previous steps) and the local config provided at runtime. This local API config is exactly same as we would expect in other APIs like `showConversation()` or `showFAQs()`.
+
+We will use this configuration for current issue as well as next issue filed in same session.
+
+<Admonition type="info" title="Note">
+
+You need to call this API after [install API](/sdkx-cocos2dx/getting-started/#initializing) and before handleProactiveLink(link)
```
Comment (jaydeep-helpshift) [inline docs/sdkx-cocos2dx/outbound-support.mdx:196]:
```suggestion
You need to call this API after [install API](/sdkx-cocos2dx/getting-started/#initializing) and before `handleProactiveLink(link)`
```

File: docs/sdkx-cocos2dx/outbound-support.mdx [hunk 26]
```diff
@@ -0,0 +1,253 @@
...
+    {
+        return false;
+    }
+    ...
+    ...
+    HelpshiftCocos2dx::setHelpshiftProactiveConfigCollector(getAPIConfig);
+}
+```
+
+<Admonition type="info" title="Note">
```
Comment (jaydeep-helpshift) [inline docs/sdkx-cocos2dx/outbound-support.mdx:249]: Can remove this admonition

File: docs/sdkx-cocos2dx/sdk-configuration.mdx
Comment (jaydeep-helpshift) [inline docs/sdkx-cocos2dx/sdk-configuration.mdx:1]: The header indentation on this page is messed up. Ensure headings have proper h1, h2, h3 etc tags.

File: docs/sdkx-cocos2dx/sdk-configuration.mdx [hunk 27]
```diff
@@ -0,0 +1,486 @@
...
+
+</Admonition>
+
+## Common configurations {#common-config}
+
+Common configurations apply to both iOS and Android platforms.
+
+### Enable Logging {#enable-logging}
+
+Upon setting `enableLogging` to `true`, Helpshift SDK logs will be generated in the Visual Studio console. These logs can be useful for debugging the SDK during integration. Turning on logging can help developers resolve common integration issues. The verbosity of the logs will change depending on whether the app is compiled for Debug or Release mode. We log more information in debug mode than in release mode.
```
Comment (jaydeep-helpshift) [inline docs/sdkx-cocos2dx/sdk-configuration.mdx:35]:
```suggestion
Upon setting `enableLogging` to `true`, Helpshift SDK logs will be generated in the Xcode/Android Studio console. These logs can be useful for debugging the SDK during integration. Turning on logging can help developers resolve common integration issues. The verbosity of the logs will change depending on whether the app is compiled for Debug or Release mode. We log more information in debug mode than in release mode.
```

File: docs/sdkx-cocos2dx/sdk-configuration.mdx [hunk 28]
```diff
@@ -0,0 +1,486 @@
...
+
+<div className="compact ">
+
+|               |                                                                                                                                  |
+| ------------- | -------------------------------------------------------------------------------------------------------------------------------- |
+| Option        | <code>enableContactUs</code>                                                                                                     |
+| Values        | <code>"ALWAYS"</code> / <code>"AFTER_VIEWING_FAQS"</code> / <code>"AFTER_MARKING_ANSWER_UNHELPFUL"</code> / <code>"NEVER"</code> |
+| Default       | <code>null</code>                                                                                                                |
+| Min SDK       | v10.2.0                                                                                                                          |
+| Supported by: | <code>ShowFAQs()</code>, <code>ShowFAQSection()</code>, <code>ShowSingleFAQ()</code>                                             |
```
Comment (jaydeep-helpshift) [inline docs/sdkx-cocos2dx/sdk-configuration.mdx:114]: API names are from Xamarin. Check for all configs.

File: docs/sdkx-cocos2dx/sdk-configuration.mdx [hunk 29]
```diff
@@ -0,0 +1,486 @@
...
+|               |                                                             |
+| ------------- | ----------------------------------------------------------- |
+| Option:       | <code>inAppNotificationAppearance</code>                    |
+| Sub-Options:  | <code>bannerBackgroundColor</code> / <code>textColor</code> |
+| Min SDK       | v10.2.0                                                     |
+| Supported by: | <code>install</code>                                        |
+
+Example:
+
+```cpp
```
Comment (jaydeep-helpshift) [inline docs/sdkx-cocos2dx/sdk-configuration.mdx:431]: Formatting for this code snippet

File: docs/sdkx-cocos2dx/sdk-configuration.mdx [hunk 30]
```diff
@@ -0,0 +1,486 @@
...
+HelpshiftCocos2dx::showConversation(config);
+```
+
+## Theming the SDK {#theming}
+
+Please check the [design](/sdkx-cocos2dx/design/) page to apply your styles and theming to SDK.
+
+## Tracking {#tracking}
+
+<!---
```
Comment (jaydeep-helpshift) [inline docs/sdkx-cocos2dx/sdk-configuration.mdx:482]: Remove the comment

File: docs/sdkx-cocos2dx/support-tools.mdx [hunk 31]
```diff
@@ -0,0 +1,143 @@
...
+**You'll need the `publish-id` of the section in this case:**
+
+<Image src="/static/books/misc/section-publish-id.png" width="full" />{" "}
+
+```cpp
+cocos2d::ValueMap faqConfig = getConfig();
+HelpshiftCocos2dx::showFAQSection(sectionId, faqConfig);
+```
+
+## Showing a Particular FAQ {#faq-view}
```
Comment (jaydeep-helpshift) [inline docs/sdkx-cocos2dx/support-tools.mdx:101]: The content arrangement here is a bit different from showFAQSection. Ensure both follow the same pattern.

File: docs/sdkx-cocos2dx/support-tools.mdx [hunk 32]
```diff
@@ -0,0 +1,143 @@
...
+
+</Centered>
+
+Example :
+
+```cpp
+cocos2d::ValueMap getConfig() {
+    cocos2d::ValueMap configMap;
+
+    ValueVector tags;
```
Comment (jaydeep-helpshift) [inline docs/sdkx-cocos2dx/support-tools.mdx:47]: Don't think we should show code for adding tags here.

File: docs/sdkx-cocos2dx/tracking.mdx [hunk 33]
```diff
@@ -0,0 +1,213 @@
...
+HelpshiftCocos2dx::showFAQs(config);
+```
+
+<Admonition type="info" title="Note">
+
+- Metadata should only be sent as String key-value pairs.
+- Once customMetadata is set, if you want to reset it, then you’ll have to call the SDK X API with an empty map, i.e.
+
+```cpp
+ValueMap config;
```
Comment (jaydeep-helpshift) [inline docs/sdkx-cocos2dx/tracking.mdx:80]: Won't this be null rather than empty? 🤔

File: docs/sdkx-cocos2dx/tracking.mdx [hunk 34]
```diff
@@ -0,0 +1,213 @@
...
+HelpshiftCocos2dx::showFAQs(config);
+```
+
+</Admonition>
+
+## Attaching Custom Issue Fields to conversations {#set-custom-issue-fields}
+
+You can attach Custom Issue Fields to every new conversation started by the user. A Custom Issue Field should have a key, a datatype, and a value. The Helpshift SDK allows the addition of Custom Issue Fields by using the `cifs` key in API config map.
+
+<Admonition type="info" title="Note">
```
Comment (jaydeep-helpshift) [inline docs/sdkx-cocos2dx/tracking.mdx:92]: Can remove the admonition

File: docs/sdkx-cocos2dx/tracking.mdx (hunk 16 above)
Comment (jaydeep-helpshift) [reply docs/sdkx-cocos2dx/tracking.mdx:98]: It's the same way in all docs.

File: docs/sdkx-cocos2dx/users.mdx [hunk 35]
```diff
@@ -0,0 +1,301 @@
...
+
+- In order to create a logged-in user in Helpshift, the use of either `userId` or `userEmail` is required.
+- The `userId` or `userEmail` should uniquely identify the user across all users of your app. It should **NOT** be duplicated for two or more different users.
+- If the login API is called with different user identifier, then it first logs out the currently logged-in user and then logs in with this user identifier.
+- It is best to call login immediately when your app user logs in.
+- If you are setting `fullPrivacy` to true, you shouldn't use email as the only identifier in the login API for that user. This will result in the creation of an anonymous user.
+
+Example:
+
+```cpp
```
Comment (jaydeep-helpshift) [inline docs/sdkx-cocos2dx/users.mdx:124]: Check the corresponding code snippet in native iOS/Android/Xamarin and follow the same structure here. This change was done after work on cocos docs started.

File: docs/sdkx-cocos2dx/users.mdx [hunk 36]
```diff
@@ -0,0 +1,301 @@
...
+<Admonition type="info" title="Note">
+
+- If the secret key is regenerated for an app on the Dashboard, you should ensure that you update your endpoint code as well, in order to generate a valid 'user authentication token'. After this, your Admins should delete the old secret key from the Dashboard in order to ensure requests using a 'user authentication token' generated using the old secret key become invalid.
+- If you are setting fullPrivacy flag to true, the 'user authentication token' should be generated only on the user ID.
+
+</Admonition>
+
+Example:
+
+```cpp
```
Comment (jaydeep-helpshift) [inline docs/sdkx-cocos2dx/users.mdx:196]: Check the corresponding code snippet in native iOS/Android/Xamarin and follow the same structure here. This change was done after work on cocos docs started.

File: src/data/homeSections.json
Comment (jaydeep-helpshift) [inline src/data/homeSections.json:1]: This seems like a merge conflict resolution gone wrong. Don't replace/remove values for any other platforms. Just ensure you are adding values for cocos.

**Review by jaydeep-helpshift — CHANGES_REQUESTED** [review]

**Review by sumeet-helpshift — APPROVED** [review]

File: docs/sdkx-cocos2dx/getting-started.mdx [hunk 37]
```diff
@@ -0,0 +1,308 @@
...
+  classes="mb-4"
+/>
+
+## Step 2 - Add Helpshift Plugin to your Cocos2dx project {#add-to-project}
+
+<Tabs groupId="platform">
+
+<TabItem value="android" label="Android">
+
+1. Copy the `android/src/com/helpshift` folder to the `src/com/` folder of your Android project.
```
[the remark this answers is not in the document]
Comment (helpshift-tanuj) [reply docs/sdkx-cocos2dx/getting-started.mdx:38]: @jaydeep-helpshift  `src/com` is inside `proj.android` or `proj.android-studio` directory, thats why i metioned android project, Will change it to android project of your cocos game.

File: docs/sdkx-cocos2dx/support-tools.mdx [hunk 38]
```diff
@@ -0,0 +1,124 @@
...
+Example :
+
+```cpp
+cocos2d::ValueMap faqConfig = getConfig();
+HelpshiftCocos2dx::showFAQs(faqConfig);
+```
+
+## Showing a Particular FAQ Section {#faq-section-view}
+
+You can use the API call `HelpshiftCocos2dx::showFAQSection(sectionId, configMap)` to invoke a particular section of your FAQs with its FAQ section id. Checkout more config options[API Options](/sdkx-cocos2dx/sdk-configuration/).
```
Comment (helpshift-tanuj) [inline docs/sdkx-cocos2dx/support-tools.mdx:80]: Space between options and [API Options]

File: docs/sdkx-cocos2dx/support-tools.mdx (hunk 31 above)
Comment (helpshift-tanuj) [reply docs/sdkx-cocos2dx/support-tools.mdx:101]: Will fix this

File: docs/sdkx-cocos2dx/support-tools.mdx (hunk 32 above)
Comment (helpshift-tanuj) [reply docs/sdkx-cocos2dx/support-tools.mdx:47]: @jaydeep-helpshift I added just to add something in config, Will try to make it generic.

**Review by IamJoyson — APPROVED** [review]

File: docs/sdkx-cocos2dx/getting-started.mdx [hunk 39]
```diff
@@ -0,0 +1,308 @@
...
+</TabItem>
+
+</Tabs>
+
+</TabItem>
+<TabItem value="ios" label="iOS">
+
+1. Unzip the package, add the contents of the `ios/Classes` folder into your Xcode project.
+2. From `ios/sdk` folder drag & drop `Helpshift.xcframework` into the Frameworks, Libraries and Embedded content in the General section of your Xcode project.
+3. If your game does not contain any swift code, set `Always Embed Swift Standard Libaries` under `Build Settings` to YES.
```
[the remark this answers is not in the document]
Comment (IamJoyson) [reply docs/sdkx-cocos2dx/getting-started.mdx:124]: We need this step, it was documented in the legacy SDK. I tested it will give errors.

File: docs/sdkx-cocos2dx/getting-started.mdx [hunk 40]
```diff
@@ -0,0 +1,308 @@
...
+4. In the Build Settings of your Xcode project, add `-fmodules` and `-fcxx-modules` flags under the `Other C Flags` section in both Debug and Release. Be careful while adding these flags in Debug and Release, because if you override the default flags you might face build errors. The `Other C Flags` section would look this after adding mentioned flags:
```
[the remark this answers is not in the document]
Comment (IamJoyson) [reply docs/sdkx-cocos2dx/getting-started.mdx:125]: IN our plugin file we are importing it as @import HelpshiftX; so when users add our plugin file they should it to project settings.

File: docs/sdkx-cocos2dx/migration-guide.mdx [hunk 41]
```diff
@@ -0,0 +1,635 @@
...
+
+### Helpshift User Management APIs
+
+Replace the following APIs for user login/logout -
+
+#### Login
+
+Replace legacy SDK login code -
+
+```cpp
```
[the remark this answers is not in the document]
Comment (IamJoyson) [reply docs/sdkx-cocos2dx/migration-guide.mdx:198]: It is the formatter which enforces the extra line before code snipped

File: docs/sdkx-cocos2dx/deep-linking.mdx [hunk 42]
```diff
@@ -0,0 +1,136 @@
...
+
+<Admonition type="info" title="Note">
+
+All the public APIs in the SDK should be called after initializing the SDK via [install API](/sdkx-cocos2dx/getting-started#initializing)
+
+</Admonition>
+
+Use custom URL schemes to direct users to particular link in your app
+from Helpshift FAQs. You can use these schemes to provide a more
+seamless experience for the user. For example, if your cocos app has a
```
Comment (helpshift-ps) [inline docs/sdkx-cocos2dx/deep-linking.mdx:30]:
```suggestion
seamless experience for the user. For example, if your Cocos app has a
```

File: docs/sdkx-cocos2dx/migration-guide.mdx [hunk 43]
```diff
@@ -0,0 +1,640 @@
...
+
+We recommend to hold off on SDK X migration if you are currently using the following APIs/features -
+
+- [Guided issue filing](/cocos2d-x/support-tools-android/#dynamic-forms)
+- [FAQ filtering by tags](/cocos2d-x/support-tools-android/#faq-filtering)
+- [Minimum Issue description length](/cocos2d-x/sdk-configuration-android/#minimum-issue-desc-length)
+- [SDK UI strings customization](/cocos2d-x/string-customization-android/)
+- [Switching between light and dark themes](/cocos2d-x/design-android/#skinning)
+
+These features are yet to be developed, and we may add them to our future roadmap. You can reach out to [support@helpshift.com](mailto:support@helpshift.com) to raise any query for the above.
```
Comment (helpshift-ps) [inline docs/sdkx-cocos2dx/migration-guide.mdx:57]: You can remove this line based on the above comment. Check this for reference - https://developers.helpshift.com/sdkx-xamarin/migration-guide/
